.doom.d/config.org
#+TITLE:   Config
#+AUTHOR:  Caio Borges Aguida Geraldes (@silenus32)
#+EMAIL:   caioaguida@protonmail.com
#      _ _
#  ___(_) | ___ _ __  _   _ ___
# / __| | |/ _ \ '_ \| | | / __|
# \__ \ | |  __/ | | | |_| \__ \
# |___/_|_|\___|_| |_|\__,_|___/
#


* Information

#+BEGIN_SRC elisp
;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Caio Borges Aguida Geraldes"
      user-mail-address "caioaguida@gmail.com")
#+END_SRC

* Themes, fonts
** Fonts

The configuration bellow intends to present a workspace for a historical linguist. The package unicode-fonts (loaded at [[file:packages.el::unicode-fonts][packages.el]]) allows the diacritics and non-standard scripts to be properly displayed at the editor. To my knowledge, the only font that encompasses all the diacritics and scripts I use in my texts is Noto Sans Mono. I chose to use the version patched in the package [[https://www.nerdfonts.com/][Nerd Fonts]], the size is arbitrary.

#+BEGIN_SRC elisp
(if (daemonp)
    (add-hook 'after-make-frame-functions
        (lambda (frame)
            (with-selected-frame frame
            (require 'unicode-fonts)
            (unicode-fonts-setup)))))
  (require 'unicode-fonts)
  (unicode-fonts-setup)

  (setq doom-font (font-spec :family "NotoSansMono Nerd Font" :size 14))

#+END_SRC

** Theme
#+BEGIN_SRC elisp
(if (daemonp)
    (add-hook 'after-make-frame-functions
        (lambda (frame)
            (with-selected-frame frame
            (set-frame-parameter (selected-frame) 'alpha '(90 . 90))
            (add-to-list 'default-frame-alist '(alpha . (90 . 90)))))))

(use-package ewal
  :init (setq ewal-use-built-in-always-p nil
              ewal-use-built-in-on-failure-p t
              ewal-built-in-palette "gruvbox"
              ewal-json-file "/home/silenux/.cache/wal"))
(use-package ewal-doom-themes
  :init (progn
          (setq doom-theme-underline-parens t
                my:rice:font (font-spec
                              :family "NotoSansMono Nerd Font"
                              :weight 'semi-bold
                              :size 11.0))
          (show-paren-mode +1)
          (global-hl-line-mode)
          (set-frame-font my:rice:font nil t)
          (add-to-list  'default-frame-alist
                        `(font . ,(font-xlfd-name my:rice:font))))
    :config (progn
            (load-theme   'ewal-doom-vibrant t)
            (enable-theme 'ewal-doom-vibrant)))

 (defun toggle-transparency ()
   (interactive)
   (let ((alpha (frame-parameter nil 'alpha)))
     (set-frame-parameter
      nil 'alpha
      (if (eql (cond ((numberp alpha) alpha)
                     ((numberp (cdr alpha)) (cdr alpha))
                     ;; Also handle undocumented (<active> <inactive>) form.
                     ((numberp (cadr alpha)) (cadr alpha)))
               100)
          '(95 . 50) '(100 . 100)))))
 (global-set-key (kbd "C-c t") 'toggle-transparency)

#+END_SRC

* Custom Variables

#+BEGIN_SRC elisp
;; (custom-set-variables)
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 ;; '(ansi-color-names-vector
   ;; ["#21242b" "#ff6c6b" "#98be65" "#ECBE7B" "#51afef" "#c678dd" "#46D9FF" "#bbc2cf"])
 ;; '(fci-rule-color "#5B6268")
 ;; '(jdee-db-active-breakpoint-face-colors (cons "#1B2229" "#51afef"))
 ;; '(jdee-db-requested-breakpoint-face-colors (cons "#1B2229" "#98be65"))
 ;; '(jdee-db-spec-breakpoint-face-colors (cons "#1B2229" "#3f444a"))
 ;; '(objed-cursor-color "#ff6c6b")

 ;; '(package-selected-packages (quote (doom-themes)))
 ;; '(pdf-view-midnight-colors (cons "#bbc2cf" "#282c34"))
 ;; '(rustic-ansi-faces
 ;;   ["#282c34" "#ff6c6b" "#98be65" "#ECBE7B" "#51afef" "#c678dd" "#46D9FF" "#bbc2cf"])
 ;; '(vc-annotate-background "#282c34")
 ;; '(vc-annotate-color-map
 ;;   (list
 ;;    (cons 20 "#98be65")
 ;;    (cons 40 "#b4be6c")
 ;;    (cons 60 "#d0be73")
 ;;    (cons 80 "#ECBE7B")
 ;;    (cons 100 "#e6ab6a")
 ;;    (cons 120 "#e09859")
 ;;    (cons 140 "#da8548")
 ;;    (cons 160 "#d38079")
 ;;    (cons 180 "#cc7cab")
 ;;    (cons 200 "#c678dd")
 ;;    (cons 220 "#d974b7")
 ;;    (cons 240 "#ec7091")
 ;;    (cons 260 "#ff6c6b")
 ;;    (cons 280 "#cf6162")
 ;;    (cons 300 "#9f585a")
 ;;    (cons 320 "#6f4e52")
 ;;    (cons 340 "#5B6268")
 ;;    (cons 360 "#5B6268")))
 ;; '(vc-annotate-very-old-color nil))
;; (custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 ;; )

#+END_SRC

* LaTeX

#+BEGIN_SRC elisp

;; Latex configuration
;;
(setq TeX-auto-save t)
(setq TeX-parse-self t)
(setq-default TeX-master nil)

(add-hook 'LaTeX-mode-hook 'visual-line-mode)
(add-hook 'LaTeX-mode-hook 'flyspell-mode)
(add-hook 'LaTeX-mode-hook 'LaTeX-math-mode)

(add-hook 'LaTeX-mode-hook 'turn-on-reftex)
(setq reftex-plug-into-AUCTeX t)

(setq +latex-viewers '(zathura))
(setq-default TeX-engine 'xetex)
(setq-default TeX-PDF-mode t)

#+END_SRC

* R with ESS

Defines a key binding to open an R terminal and maps "_" (underscore) to " <- ".

#+BEGIN_SRC elisp
(global-set-key (kbd "C-c r") 'run-ess-r)
(use-package ess-r-mode
  :bind
  (:map ess-r-mode-map
        ("_" . ess-insert-assign))
  (:map inferior-ess-r-mode-map
        ("_" . ess-insert-assign)))
#+END_SRC

* Neotree
#+begin_src emacs-lisp
;; overwrite the default hidden file filter, so it uses .gitignore
        (defun neo-util--hidden-path-filter (node)
            "it reads each nodule on the list, and determines if its ignored by git."
            (if neo-buffer--show-hidden-file-p
                ;; all files should be shown
                node
                ;; hiding is enabled, use git check-ignore to determeine which to show
                (if
                    ;; if the output is empty (file should be shown) return the node
                    (string=
                        (string-trim
                            (shell-command-to-string
                                (format
                                    "git -C %s check-ignore %s"
                                    (file-name-directory node)
                                    node
                                )
                            )
                        )
                        ""
                    )
                    node
                    ;; git outputed something, file shold be hidden
                    nil
                )
            )
        )
#+end_src
* Other packages

** Elfeed

These create the bindings:

#+BEGIN_SRC elisp
(global-set-key (kbd "C-x f") 'elfeed)
#+END_SRC

And sets a couple of RSS feeds for me:
- *Notícias (News)*
  - [[www.theguardian.com][Guardian]]
    - [[https://www.theguardian.com/uk/culture/rss][Guardian Culture]]
    - [[https://www.theguardian.com/uk/world/americas/rss][Guardian World - Americas]]
  - [[https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Ffeeds.folha.uol.com.br%2Filustrissima%2Frss091.xml][Folha Ilustríssima]]
  - [[https://feedly.com/i/subscription/feed%2Fhttp%3A%2F%2Fpiaui.folha.uol.com.br%2Ffeed%2F][Piauí]]
  - [[https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Fwww.nexojornal.com.br%2Frss.xml][Nexo]]
- *Linguística (Linguistics)*
  - [[https://linguistlist.org/issues/rss/mostrecent.xml][The Linguist List]]
  - [[https://www.languagesciences.cam.ac.uk/taxonomy/term/84/feed][Cambridge Linguistic Sciences]]
  - [[https://www.degruyter.com/journalissuetocrss/journals/indo/indo-overview.xml][Indogermanische Forschungen]]
  - [[https://www.mitpressjournals.org/action/showFeed?jc=coli&type=etoc&feed=rss][Computational Linguistics (Journal)]]
  - [[https://indology.info/rss1.xml][Indology]]

#+BEGIN_SRC elisp
(setq elfeed-feeds
      '(("https://www.theguardian.com/uk/culture/rss" news culture)
        ("https://www.theguardian.com/world/americas/rss" news)
        ("https://feedly.com/i/subscription/feed/https://piaui.folha.uol.com.br/feed" news)
        ("https://feedly.com/i/subscription/feed/https://www.nexojornal.com.br/feed" news)
        ("https://linguistlist.org/issues/rss/mostrecent.xml" linguistics)
        ("https://www.languagesciences.cam.ac.uk/taxonomy/term/84/feed" linguistics)
        ("https://www.degruyter.com/journalissuetocrss/journals/indo/indo-overview.xml" linguistics philology academic-stuff)
        ("https://www.mitpressjournals.org/action/showFeed?jc=coli&type=etoc&feed=rss" linguistics)
        ("https://www.oxfordscholarship.com/newsrss" academic-stuff)
        ("https://blog.philsoc.org.uk/feed" philology linguistics academic-stuff)
        ("https://indology.info/rss1.xml" linguistics indology philology academic-stuff)
        ("https://consultingphilologist.wordpress.com/" philology academic-stuff linguistics)
        ("https://crewsproject.wordpress.com/" philology academic-stuff linguistics)
        ("https://escamandro.wordpress.com/feed" literature others)
        ("https://toujourmicelio.wordpress.com/feed" philosophy others)))
#+END_SRC

** PIE-MACS

#+BEGIN_SRC elisp
;; Custom Input Methods
(let ((default-directory  "~/.doom.d/lisp/"))
  (normal-top-level-add-subdirs-to-load-path))

(require 'pie-macs)
#+END_SRC

* Email Configuration

#+BEGIN_SRC elisp
;; mu4e config
(require 'mu4e)

;; use mu4e for e-mail in emacs
(setq mail-user-agent 'mu4e-user-agent)

(setq mu4e-drafts-folder "/[Gmail].Rascunhos")
(setq mu4e-sent-folder "/[Gmail].E-mails enviados")
(setq mu4e-trash-folder "/[Gmail].Lixeira")

;; don't save message to Sent Messages, Gmail/IMAP takes care of this
(setq mu4e-sent-messages-behavior 'delete)

;; (See the documentation for `mu4e-sent-messages-behavior' if you have
;; additional non-Gmail addresses and want assign them different
;; behavior.)

;; setup some handy shortcuts
;; you can quickly switch to your Inbox -- press ``ji''
;; then, when you want archive some messages, move them to
;; the 'All Mail' folder by pressing ``ma''.

(setq mu4e-maildir-shortcuts
    '( (:maildir "/INBOX" :key ?i)
       (:maildir "/[Gmail].E-mails enviados" :key ?s)
       (:maildir "/[Gmail].Lixeira" :key ?t)
       (:maildir "/[Gmail].Todos os e-mails" :key ?a)))

;; allow for updating mail using 'U' in the main view:
(setq mu4e-get-mail-command "offlineimap")

;; sending mail -- replace USERNAME with your gmail username
;; also, make sure the gnutls command line utils are installed
;; package 'gnutls-bin' in Debian/Ubuntu

(require 'smtpmail)
(setq message-send-mail-function 'smtpmail-send-it
   starttls-use-gnutls t
   smtpmail-starttls-credentials '(("smtp.gmail.com" 587 nil nil))
   smtpmail-auth-credentials
     '(("smtp.gmail.com" 587 "caioaguida@gmail.com" nil))
   smtpmail-default-smtp-server "smtp.gmail.com"
   smtpmail-smtp-server "smtp.gmail.com"
   smtpmail-smtp-service 587)

;; alternatively, for emacs-24 you can use:
;;(setq message-send-mail-function 'smtpmail-send-it
;;     smtpmail-stream-type 'starttls
;;     smtpmail-default-smtp-server "smtp.gmail.com"
;;     smtpmail-smtp-server "smtp.gmail.com"
;;     smtpmail-smtp-service 587)

;; don't keep message buffers around
(setq message-kill-buffer-on-exit t)

#+END_SRC


.w3m/history
https://www1.folha.uol.com.br/colunas/monicabergamo/2019/10/homem-de-esporte-grosseiro-pode-tratar-naturalmente-um-filho-gay-diz-popo.shtml
file:///home/silenux/Documentos/Notebooks/CaseAttrac/rnotebook.html
http://gen.lib.rus.ec/search.php?req=The+Oxford+Reference+Guide+to+Lexical+Functional+Grammar&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=Guide+to+Lexical+Functional+Grammar&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=Dalrymple&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=Dalrymple+Syntax&open=0&res=25&view=simple&phrase=1&column=def
http://185.39.10.101/ads.php?md5=25ECDF60311A2CE0F1DEB72B74766B0E
http://93.174.95.29/_ads/25ECDF60311A2CE0F1DEB72B74766B0E
http://93.174.95.29/main/289000/25ecdf60311a2ce0f1deb72b74766b0e/%28Syntax%20and%20Semantics%2C%20Volume%2034%20Syntax%20and%20Semantics%20Syntax%20and%20Semantics%29%20Mary%20Dalrymple%20-%20Lexical%20functional%20grammar-AP%20%282001%29.djvu
https://www1.folha.uol.com.br/colunas/monicabergamo/2019/10/eduardo-bolsonaro-diz-que-se-esquerda-radicalizar-resposta-pode-ser-um-novo-ai-5.shtml
http://gen.lib.rus.ec/search.php?req=devine&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=A.+M.+Devine&column=author
http://gen.lib.rus.ec/search.php?req=a+m+devine&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://93.174.95.29/_ads/919DD1939A35DD9CA00394C082A8B7DA
http://93.174.95.29/main/310000/919dd1939a35dd9ca00394c082a8b7da/A.%20M.%20Devine%2C%20Laurence%20D.%20Stephens%20-%20Latin%20Word%20Order_%20Structured%20Meaning%20and%20Information-Oxford%20University%20Press%2C%20USA%20%282006%29.pdf
http://93.174.95.29/_ads/906AB8800369060464E78194C9E372D5
http://185.39.10.101/ads.php?md5=906AB8800369060464E78194C9E372D5
http://b-ok.cc/md5/906AB8800369060464E78194C9E372D5
http://libgen.pw/item/detail/id/531591
http://bookfi.net/md5/906AB8800369060464E78194C9E372D5
http://bookfi.net/dl/1152212/8d67ac
http://gen.lib.rus.ec/search.php?req=nltk&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/search.php?&req=nltk&phrase=1&view=simple&column=def&sort=year&sortmode=DESC
http://185.39.10.101/ads.php?md5=14A19ED32CD67FA00DACAC1C27590DCA
http://booksdl.org/get.php?md5=14a19ed32cd67fa00dacac1c27590dca&key=7D105KQ9HSAUTBTL
http://185.39.10.101/ads.php?md5=9294ACA47E57263B9AE254BEC9AD2C1B
http://booksdl.org/get.php?md5=9294aca47e57263b9ae254bec9ad2c1b&key=IJE3890J3Q974UQB
http://185.39.10.101/ads.php?md5=C453736B6CA4E3344983FD9A29A7B537
http://booksdl.org/get.php?md5=c453736b6ca4e3344983fd9a29a7b537&key=ZALB7LVRWA7GWPOZ
http://gen.lib.rus.ec/search.php?req=+Letters+Against+the+Firmament+&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/foreignfiction/index.php?s=%20Letters%20Against%20the%20Firmament%20&f_lang=All&f_columns=0&f_ext=All&f_group=1
http://gen.lib.rus.ec/search.php?req=Marcel+Bizos&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=lesbos&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=lesbian&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=lesbian+dia&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=lesbian+greek&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=lesbian+dialect&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=aeolic&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=aeolisch&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=aeolische&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=%C3%A4olische&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=Die+aiolischen+Dialekte&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://93.174.95.29/_ads/16FD0B63DC84E86E3678EC08E031069F
http://93.174.95.29/main/728000/16fd0b63dc84e86e3678ec08e031069f/Wolfgang%20Blumel%20-%20Die%20aiolischen%20Dialekte_%20Phonologie%20und%20Morphologie%20der%20inschriftlichen%20Texte%20aus%20generativer%20Sicht%20%28Erganzungshefte%20zur%20Zeitschrift%20fur%20vergleichende%20Sprachforschung%29%20%20-Vandenhoeck%20.pdf
http://gen.lib.rus.ec/search.php?req=%C3%A9olien&open=0&res=25&view=simple&phrase=1&column=def
http://gen.lib.rus.ec/search.php?req=Garc%C3%ADa-Ram%C3%B3n&column=author
http://gen.lib.rus.ec/search.php?req=Jos%C3%A9+Garc%C3%ADa-Ram%C3%B3n&open=0&res=25&view=simple&phrase=1&column=author
http://gen.lib.rus.ec/search.php?req=The+Luwians&lg_topic=libgen&open=0&view=simple&res=25&phrase=1&column=def
http://185.39.10.101/ads.php?md5=9FFC1390F83AD6698D90880AE2F071AE
http://b-ok.cc/md5/9FFC1390F83AD6698D90880AE2F071AE
https://libgen.pw/item?id=518828
http://93.174.95.29/_ads/9FFC1390F83AD6698D90880AE2F071AE
http://93.174.95.29/main/518000/9ffc1390f83ad6698d90880ae2f071ae/H.%20Craig%20Melchert%20-%20The%20Luwians%20%28Handbook%20of%20Oriental%20Studies%20Handbuch%20Der%20Orientalistik%29%20%282003%29.pdf
https://www1.folha.uol.com.br/cotidiano/2020/01/como-cai-no-golpe-que-sequestra-a-conta-do-whatsapp-para-extorquir-dinheiro-dos-contatos-telefonicos.shtml
https://www1.folha.uol.com.br/mercado/2020/01/novo-reajuste-do-salario-minimo-cria-impasse-juridico-no-governo.shtml
https://www.wsj.com/
https://www.wsj.com/articles/warren-sanders-clash-ahead-of-democratic-debate-11578970442?mod=article_inline
https://www.wsj.com/news/life-arts/arts
https://www.wsj.com/articles/square-toed-shoes-the-egregious-90s-style-returns-11578516098
https://www.ft.com/
https://www.ft.com/world/americas
https://www.ft.com/content/206a3598-3333-11ea-9703-eea0cae3f0de

.doom.d/config.el
;; overwrite the default hidden file filter, so it uses .gitignore
        (defun neo-util--hidden-path-filter (node)
            "it reads each nodule on the list, and determines if its ignored by git."
            (if neo-buffer--show-hidden-file-p
                ;; all files should be shown
                node
                ;; hiding is enabled, use git check-ignore to determeine which to show
                (if
                    ;; if the output is empty (file should be shown) return the node
                    (string=
                        (string-trim
                            (shell-command-to-string
                                (format
                                    "git -C %s check-ignore %s"
                                    (file-name-directory node)
                                    node
                                )
                            )
                        )
                        ""
                    )
                    node
                    ;; git outputed something, file shold be hidden
                    nil
                )
            )
        )

;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Caio Borges Aguida Geraldes"
      user-mail-address "caioaguida@gmail.com")

(if (daemonp)
    (add-hook 'after-make-frame-functions
        (lambda (frame)
            (with-selected-frame frame
            (require 'unicode-fonts)
            (unicode-fonts-setup)))))
  (require 'unicode-fonts)
  (unicode-fonts-setup)

  (setq doom-font (font-spec :family "NotoSansMono Nerd Font" :size 14))

(if (daemonp)
    (add-hook 'after-make-frame-functions
        (lambda (frame)
            (with-selected-frame frame
            (set-frame-parameter (selected-frame) 'alpha '(90 . 90))
            (add-to-list 'default-frame-alist '(alpha . (90 . 90)))))))

(set-face-attribute 'flycheck-error nil :underline '(:color "Red" :style wave))
(use-package ewal
  :init (setq ewal-use-built-in-always-p nil
              ewal-use-built-in-on-failure-p t
              ewal-built-in-palette "gruvbox"
              ewal-json-file "/home/silenux/.cache/wal"))
(use-package ewal-doom-themes
  :init (progn
          (setq doom-theme-underline-parens t
                my:rice:font (font-spec
                              :family "NotoSansMono Nerd Font"
                              :weight 'semi-bold
                              :size 11.0))
          (show-paren-mode +1)
          (global-hl-line-mode)
          (set-frame-font my:rice:font nil t)
          (add-to-list  'default-frame-alist
                        `(font . ,(font-xlfd-name my:rice:font))))
    :config (progn
            (load-theme   'ewal-doom-vibrant t)
            (enable-theme 'ewal-doom-vibrant)))

 (defun toggle-transparency ()
   (interactive)
   (let ((alpha (frame-parameter nil 'alpha)))
     (set-frame-parameter
      nil 'alpha
      (if (eql (cond ((numberp alpha) alpha)
                     ((numberp (cdr alpha)) (cdr alpha))
                     ;; Also handle undocumented (<active> <inactive>) form.
                     ((numberp (cadr alpha)) (cadr alpha)))
               100)
          '(95 . 50) '(100 . 100)))))
 (global-set-key (kbd "C-c t") 'toggle-transparency)

;; (custom-set-variables)
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 ;; '(ansi-color-names-vector
   ;; ["#21242b" "#ff6c6b" "#98be65" "#ECBE7B" "#51afef" "#c678dd" "#46D9FF" "#bbc2cf"])
 ;; '(fci-rule-color "#5B6268")
 ;; '(jdee-db-active-breakpoint-face-colors (cons "#1B2229" "#51afef"))
 ;; '(jdee-db-requested-breakpoint-face-colors (cons "#1B2229" "#98be65"))
 ;; '(jdee-db-spec-breakpoint-face-colors (cons "#1B2229" "#3f444a"))
 ;; '(objed-cursor-color "#ff6c6b")

 ;; '(package-selected-packages (quote (doom-themes)))
 ;; '(pdf-view-midnight-colors (cons "#bbc2cf" "#282c34"))
 ;; '(rustic-ansi-faces
 ;;   ["#282c34" "#ff6c6b" "#98be65" "#ECBE7B" "#51afef" "#c678dd" "#46D9FF" "#bbc2cf"])
 ;; '(vc-annotate-background "#282c34")
 ;; '(vc-annotate-color-map
 ;;   (list
 ;;    (cons 20 "#98be65")
 ;;    (cons 40 "#b4be6c")
 ;;    (cons 60 "#d0be73")
 ;;    (cons 80 "#ECBE7B")
 ;;    (cons 100 "#e6ab6a")
 ;;    (cons 120 "#e09859")
 ;;    (cons 140 "#da8548")
 ;;    (cons 160 "#d38079")
 ;;    (cons 180 "#cc7cab")
 ;;    (cons 200 "#c678dd")
 ;;    (cons 220 "#d974b7")
 ;;    (cons 240 "#ec7091")
 ;;    (cons 260 "#ff6c6b")
 ;;    (cons 280 "#cf6162")
 ;;    (cons 300 "#9f585a")
 ;;    (cons 320 "#6f4e52")
 ;;    (cons 340 "#5B6268")
 ;;    (cons 360 "#5B6268")))
 ;; '(vc-annotate-very-old-color nil))
;; (custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 ;; )

;; Latex configuration
;;
(setq TeX-auto-save t)
(setq TeX-parse-self t)
(setq-default TeX-master nil)

(add-hook 'LaTeX-mode-hook 'visual-line-mode)
(add-hook 'LaTeX-mode-hook 'flyspell-mode)
(add-hook 'LaTeX-mode-hook 'LaTeX-math-mode)

(add-hook 'LaTeX-mode-hook 'turn-on-reftex)
(setq reftex-plug-into-AUCTeX t)

(setq +latex-viewers '(zathura))
(setq-default TeX-engine 'xetex)
(setq-default TeX-PDF-mode t)

(global-set-key (kbd "C-c r") 'run-ess-r)
(use-package ess-r-mode
  :bind
  (:map ess-r-mode-map
        ("_" . ess-insert-assign))
  (:map inferior-ess-r-mode-map
        ("_" . ess-insert-assign)))

(global-set-key (kbd "C-x f") 'elfeed)

(setq elfeed-feeds
      '(("https://www.theguardian.com/uk/culture/rss" news culture)
        ("https://www.theguardian.com/world/americas/rss" news)
        ("https://feedly.com/i/subscription/feed/https://piaui.folha.uol.com.br/feed" news)
        ("https://feedly.com/i/subscription/feed/https://www.nexojornal.com.br/feed" news)
        ("https://linguistlist.org/issues/rss/mostrecent.xml" linguistics)
        ("https://www.languagesciences.cam.ac.uk/taxonomy/term/84/feed" linguistics)
        ("https://www.degruyter.com/journalissuetocrss/journals/indo/indo-overview.xml" linguistics philology academic-stuff)
        ("https://www.mitpressjournals.org/action/showFeed?jc=coli&type=etoc&feed=rss" linguistics)
        ("https://www.oxfordscholarship.com/newsrss" academic-stuff)
        ("https://blog.philsoc.org.uk/feed" philology linguistics academic-stuff)
        ("https://indology.info/rss1.xml" linguistics indology philology academic-stuff)
        ("https://consultingphilologist.wordpress.com/" philology academic-stuff linguistics)
        ("https://crewsproject.wordpress.com/" philology academic-stuff linguistics)
        ("https://escamandro.wordpress.com/feed" literature others)
        ("https://toujourmicelio.wordpress.com/feed" philosophy others)))

;; Custom Input Methods
(let ((default-directory  "~/.doom.d/lisp/"))
  (normal-top-level-add-subdirs-to-load-path))

(require 'pie-macs)

;; mu4e config
(require 'mu4e)

;; use mu4e for e-mail in emacs
(setq mail-user-agent 'mu4e-user-agent)

(setq mu4e-drafts-folder "/[Gmail].Rascunhos")
(setq mu4e-sent-folder "/[Gmail].E-mails enviados")
(setq mu4e-trash-folder "/[Gmail].Lixeira")

;; don't save message to Sent Messages, Gmail/IMAP takes care of this
(setq mu4e-sent-messages-behavior 'delete)

;; (See the documentation for `mu4e-sent-messages-behavior' if you have
;; additional non-Gmail addresses and want assign them different
;; behavior.)

;; setup some handy shortcuts
;; you can quickly switch to your Inbox -- press ``ji''
;; then, when you want archive some messages, move them to
;; the 'All Mail' folder by pressing ``ma''.

(setq mu4e-maildir-shortcuts
    '( (:maildir "/INBOX" :key ?i)
       (:maildir "/[Gmail].E-mails enviados" :key ?s)
       (:maildir "/[Gmail].Lixeira" :key ?t)
       (:maildir "/[Gmail].Todos os e-mails" :key ?a)))

;; allow for updating mail using 'U' in the main view:
(setq mu4e-get-mail-command "offlineimap")

;; sending mail -- replace USERNAME with your gmail username
;; also, make sure the gnutls command line utils are installed
;; package 'gnutls-bin' in Debian/Ubuntu

(require 'smtpmail)
(setq message-send-mail-function 'smtpmail-send-it
   starttls-use-gnutls t
   smtpmail-starttls-credentials '(("smtp.gmail.com" 587 nil nil))
   smtpmail-auth-credentials
     '(("smtp.gmail.com" 587 "caioaguida@gmail.com" nil))
   smtpmail-default-smtp-server "smtp.gmail.com"
   smtpmail-smtp-server "smtp.gmail.com"
   smtpmail-smtp-service 587)

;; alternatively, for emacs-24 you can use:
;;(setq message-send-mail-function 'smtpmail-send-it
;;     smtpmail-stream-type 'starttls
;;     smtpmail-default-smtp-server "smtp.gmail.com"
;;     smtpmail-smtp-server "smtp.gmail.com"
;;     smtpmail-smtp-service 587)

;; don't keep message buffers around
(setq message-kill-buffer-on-exit t)
